doc.go

```diff
@@ -0,0 +1,71 @@
+/*
+SQLair is a convenience layer for SQL databases that embeds Go structs directly into SQL queries.
+
+The SQL syntax is expanded with SQLair input and output expressions which indicate parts of the query that correspond to Go structs.
+This allows the user to specify the Go structs they want in the SQL query itself whilst allowing the full power of SQL to be utilised.
+This package also provides an alternative API for reading the rows from the database.
+It does not copy database/sql, it instead improves upon it and removes inconsistencies.
+
+# Basics
+
+To read the results of a query into a struct, the columns listed in the SQL are replaced by a SQLair output expression.
+The output expressions are parsed and replaced with the column names found in the `db` tags of the struct.
+Similarly, to use a query argument directly from a Go struct the question mark is replaced by a SQLair input expression.
+This package does not parse the SQL query, only the SQLair input/output expressions found within it.
+
+The characters $ and & are used to specify input and outputs expressions respectively.
+For example, given the following tagged struct "Person":
+
+	type Person struct {
+		Name	string	`db:"name"`
+		ID	int	`db:"id"`
+		Team	string  `db:"team"`
+	}
+
+Instead of the SQL query:
+
+	SELECT name, id, team
+	FROM person
+	WHERE manager_name = ?
+
+With SQLair one would write:
+
+	SELECT &Person.*
+	FROM person
+	WHERE manager_name = $Manager.name
+
+SQLair substitutes &Person.* for all columns in the `db` tags of Person.
+The input expression, $Manager.name, instructs SQLair to pass the Name field of the struct Manager as a query argument.
```

Comment:
Same here, this seems like overexplaining it. It's clear from the example.

Reply:
Removed.